middleware/rate_limiter.go

```diff
@@ -83,23 +136,46 @@ func RateLimiterWithConfig(config RateLimiterConfig) echo.MiddlewareFunc {
 // RateLimiterMemoryStore is the built-in store implementation for RateLimiter
 type (
 	RateLimiterMemoryStore struct {
-		visitors map[string]visitor
-		mutex    sync.Mutex
-		rate     rate.Limit
-		burst    int
+		visitors    map[string]Visitor
+		mutex       sync.Mutex
+		rate        rate.Limit
+		burst       int
+		expiresIn   time.Duration
+		lastCleanup time.Time
 	}
-	visitor struct {
+	// Visitor signifies a unique user's limiter details
+	Visitor struct {
 		*rate.Limiter
 		lastSeen time.Time
 	}
 )
 
+// NewRateLimiterMemoryStore returns an instance of RateLimiterMemoryStore
+func NewRateLimiterMemoryStore(
+	rate rate.Limit,
+	burst int,
+	/* using this variadic because I don't see a better way to make it optional. suggestions are welcome */
```

Comment:
What about Rob Pike's [Self-referential functions and the design of options](https://commandcenter.blogspot.com/2014/01/self-referential-functions-and-design.html)?

I used in the past and I like it . Also you can use it to set rate and burst in the same way.